### PR TITLE
Use set when parsing repos to prevent duplicates

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

#1905 didn't solve the problem - so this is a simpler approach that is probably more robust.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: sdk/python/feast/entity.py [hunk 1]
```diff
@@ -78,7 +74,8 @@ def __init__(
         self._created_timestamp: Optional[datetime] = None
         self._last_updated_timestamp: Optional[datetime] = None
 
-        self.defined_in = get_calling_file_name(inspect.stack())
+    def __hash__(self) -> int:
+        return hash(self.name)
```
Comment (woop) [inline sdk/python/feast/entity.py:78]: How would we detect and raise an exception when users provide two objects with the same name that are different?

Comment (achals) [reply sdk/python/feast/entity.py:78]: Having two entities with the same name is a bug though, isn't it?

Comment (achals) [reply sdk/python/feast/entity.py:78]: I don't think we support any FV/FT/entity/FS having the same name right now anyway

Comment (achals) [reply sdk/python/feast/entity.py:78]: Okay, I checked the code and we verify unique names for FVs but not other objects. I don't know what the behaviour would be for `FeatureStore.get_feature_view` if names can be duplicated. Plus, feature_refs would be busted (in the case of feature views and feature tables). wdyt?

Comment (achals) [reply sdk/python/feast/entity.py:78]: My earlier comments completely misunderstood your point -fixing using `id()`.

**Review by MattDelac — APPROVED** [review]
Looks good to me

Thanks for the test, this makes it safer

I can also confirm that based on the latest commit of this PR, it works on our side 🚀 
image

File: sdk/python/tests/integration/registration/test_cli_apply_duplicated_featureview_names.py [hunk 2]
```diff
@@ -45,6 +45,49 @@ def test_cli_apply_duplicated_featureview_names() -> None:
         )
 
 
+def test_cli_apply_imported_featureview() -> None:
```
Comment (woop) [inline sdk/python/tests/integration/registration/test_cli_apply_duplicated_featureview_names.py:48]: Should we add a test that ensures that duplicate names with different contents raises an exception?

Comment (achals) [reply sdk/python/tests/integration/registration/test_cli_apply_duplicated_featureview_names.py:48]: There's already a test for such a case

Comment (achals) [reply sdk/python/tests/integration/registration/test_cli_apply_duplicated_featureview_names.py:48]: done

File: sdk/python/tests/integration/registration/test_cli_apply_duplicated_featureview_names.py [hunk 3]
```diff
@@ -45,6 +45,105 @@ def test_cli_apply_duplicated_featureview_names() -> None:
...
+            "   batch_source=driver_hourly_stats,\n"
+            "   tags={'dummy': 'true'})\n"
+        )
+
+        rc, output = runner.run_with_output(["apply"], cwd=repo_path)
+
+        assert rc != 0
+        import pdb
+
+        pdb.set_trace()
```
Comment (MattDelac) [inline sdk/python/tests/integration/registration/test_cli_apply_duplicated_featureview_names.py:141]: Should that be removed ?
